pkg/registry/generic/etcd/etcd.go

```diff
@@ -440,6 +441,33 @@ func (e *Etcd) Delete(ctx api.Context, name string, options *api.DeleteOptions)
 	return e.finalizeDelete(out, true)
 }
 
+func (e *Etcd) DeleteCollection(ctx api.Context, options *api.DeleteOptions, listOptions *unversioned.ListOptions) (runtime.Object, error) {
+	// TODO: Currently, there is no easy way to remove directory in etcd with
+	// all watch events correctly delivered.
+	// So as a temporary hack, we simply remove elements one by one.
+	// Once v3 API in etcd is ready, we should switch to it.
+	listObj, err := e.List(ctx, listOptions)
+	if err != nil {
+		return nil, err
+	}
+	items, err := meta.ExtractList(listObj)
+	if err != nil {
+		return nil, err
+	}
+	for _, item := range items {
+		accessor, err := meta.Accessor(item)
+		if err != nil {
+			return nil, err
+		}
+		// TODO: Shouldn't we copy options here?
+		// TODO: Shouldn't we read returned object?
+		if _, err := e.Delete(ctx, accessor.Name(), options); err != nil {
```

Comment:
I agree - but even without any paralellization it's 3m instead of 35m in scalability tests.
I think it's good enough for the first step - we can do further improvements later.